### PR TITLE
[ZEPPELIN-3687] Fix IndexError in python interpreter with empty input

### PR DESCRIPTION
**Pull request by oxygen311**
### What is this PR for?
If input of python or pyspark interpreter is empty (contains only comments), it will fail with `IndexError` from `zeppelin_python.py`.

### What type of PR is it?
Bug Fix

### What is the Jira issue?
[ZEPPELIN-3687](https://issues.apache.org/jira/projects/ZEPPELIN/issues/ZEPPELIN-3687)

## Before:
screen-shot-2018-08-05-at-23 29 34
screen-shot-2018-08-07-at-11 48 10

## After:
screen-shot-2018-08-07-at-15 11 452
screen-shot-2018-08-07-at-15 11 45

### Questions:
* Does the licenses files need update? No 
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
Could you add unit test ?

**Comment by oxygen311** [comment]
@zjffdu Done

File: python/src/test/java/org/apache/zeppelin/python/PythonInterpreterTest.java [hunk 1]
```diff
@@ -105,4 +106,28 @@ public void testCancelIntp() throws InterruptedException, InterpreterException {
     t.join(2000);
     assertFalse(t.isAlive());
   }
+
```
Comment (zjffdu) [inline python/src/test/java/org/apache/zeppelin/python/PythonInterpreterTest.java]: Could you move them to BasePythonInterpreterTest#testPythonBasics ?

Comment (oxygen311) [reply python/src/test/java/org/apache/zeppelin/python/PythonInterpreterTest.java]: Done

**Comment by zjffdu** [comment]
LGTM, did you configure travis ? I don't see your travis build

**Comment by oxygen311** [comment]
@zjffdu @felixcheung 
Travis tests fixed

**Pull request closed without merging**